packages/x-data-grid/src/components/GridPagination.tsx

```diff
@@ -160,24 +136,19 @@ const GridPagination = forwardRef<
 
   return (
     <GridPaginationRoot
-      component="div"
+      as={rootProps.slots.basePagination}
       count={rowCount}
       page={computedPage}
-      // TODO: Remove the cast once the type is fixed in Material UI and that the min Material UI version
-      // for x-data-grid is past the fix.
-      // Note that Material UI will not mutate the array, so this is safe.
-      rowsPerPageOptions={pageSizeOptions as MutableArray<typeof pageSizeOptions>}
+      rowsPerPageOptions={pageSizeOptions}
       rowsPerPage={paginationModel.pageSize}
       onPageChange={handlePageChange}
-      onRowsPerPageChange={handlePageSizeChange}
-      {...computedProps}
+      onRowsPerPageChange={handlePageSizeChange as any}
+      disabled={disabled}
       {...locales}
       labelDisplayedRows={wrappedLabelDisplayedRows}
-      {...props}
```

Comment:
I've had to remove `...props` in some places and it makes it imposible for users to use e.g. `rootProps.slotProps.pagination` to customize the slot, but we can't both be design-system agnostic and pass any material-ui prop.

The solution here would be for users to fork the `basePagination` slot and customize it there.

Reply:
> I've had to remove ...props in some places and it makes it imposible for users to use e.g. rootProps.slotProps.pagination to customize the slot

Why though? What is different about the `Pagination` slot?

Reply:
I meant: it's not going to be possible to pass material props via `slotProps.pagination`.

Reply:
> I meant: it's not going to be possible to pass material props via slotProps.pagination.

Ahh, only via `slotProps.basePagination`?

Reply:
The problem is that base slots are not material slots anymore, they're design-system agnostic. For some of them, it might happen that the grid's base slot API matches the material API, but it's not consistent. I have been thinking about adding something like `slotProps.raw` to all base components to pass custom props to the underlying design-system, be it material or something else.